spec/solver_spec.rb

```diff
@@ -0,0 +1,22 @@
+require_relative '../solver'
+
+describe Solver do
+  before(:all) do
+    @solver = Solver.new
+  end
+
+  it 'test factorial method' do
+    @solver.factorial(2).should eql? 2
+    @solver.factorial(3).should eql? 6
+    @solver.factorial(10).should eql? 3_628_800
+  end
```

Comment:
- [ **OPTIONAL** ] **RECOMMENDED:** I think it would be better if you add a test to your `#factorial` method of a special case - when input is 0, the result should equal 1.